imap_processing/idex/idex_l2c.py

```diff
@@ -0,0 +1,255 @@
+
+    # Get the healpix indices
+    hpix_idx = hp.ang2pix(
+        nside, nest=nested, lonlat=True, theta=longitude, phi=latitude
+    )
+
+    n_pix = hp.nside2npix(nside)
+    healpix = xr.DataArray(
+        np.arange(n_pix),
+        name=CoordNames.HEALPIX_INDEX.value,
+        dims=CoordNames.HEALPIX_INDEX.value,
+        attrs=idex_attrs.get_variable_attributes("pixel_index", check_schema=False),
+    )
+
+    # Create a histogram of the raw dust event counts for each pixel
+    counts = np.histogram(hpix_idx, bins=n_pix, range=(0, n_pix))[0]
+    # Add epoch dimension
+    counts_da = xr.DataArray(
+        counts[np.newaxis, :].astype(int),
+        name="counts",
+        dims=("epoch", CoordNames.HEALPIX_INDEX.value),
+        attrs=idex_attrs.get_variable_attributes("healpix_counts"),
+    )
+    pixel_label = xr.DataArray(
+        healpix.astype(str),
+        name="pixel_label",
+        dims="pixel_index",
+        attrs=idex_attrs.get_variable_attributes("pixel_label", check_schema=False),
+    )
+    l2c_dataset = xr.Dataset(
+        coords={CoordNames.HEALPIX_INDEX.value: healpix, "epoch": epoch_da},
+        data_vars={
+            "counts": counts_da,
+            "longitude": longitude,
+            "latitude": latitude,
+            "pixel_label": pixel_label,
+        },
+    )
+    map_attrs = {
+        "Sky_tiling_type": SkyTilingType.HEALPIX.value,
+        "HEALPix_nside": nside,
+        "HEALPix_nest": nested,
+        "Spice_reference_frame": IDEX_EVENT_REFERENCE_FRAME,
+        "num_points": n_pix,
+    } | idex_attrs.get_global_attributes("imap_idex_l2c_sci-healpix")
+    l2c_dataset.attrs.update(map_attrs)
+
+    return l2c_dataset
+
+
+def idex_rectangular_map(
+    l1b_dataset: xr.Dataset,
+    epoch_da: xr.DataArray,
+    idex_attrs: ImapCdfAttributes,
+    spacing_deg: int = IDEX_SPACING_DEG,
+) -> xr.Dataset:
+    """
+    Create a rectangular map out of a l1b dataset.
+
+    Parameters
+    ----------
+    l1b_dataset : xarray.Dataset
+        IDEX L2b dataset.
+    epoch_da : xarray.DataArray
+        Epoch data array of size (1,).
+    idex_attrs : ImapCdfAttributes
+        The attribute manager for this data level.
+    spacing_deg : int
+        The spacing in degrees for the rectangular grid.
+
+    Returns
+    -------
+    map : xarray.Dataset
+        Spatially binned dust counts in a rectangular map format.
+    """
+    # Get the rectangular grid with the specified spacing
+    grid = AzElSkyGrid(spacing_deg)
+    # Make sure longitude values are in the range [0, 360)
+    longitude_wrapped = np.mod(l1b_dataset["longitude"], 360)
+    latitude = l1b_dataset["latitude"]
+    # Create a 2d histogram of the raw dust event counts for each pixel using the grid
+    # bin edges
+    counts, _, _ = np.histogram2d(
+        longitude_wrapped, latitude, bins=[grid.az_bin_edges, grid.el_bin_edges]
+    )
+    counts_da = xr.DataArray(
+        counts[np.newaxis, :, :].astype(int),
```

Comment:
Counts should generally be uint as they can't be negative and you can probably reduce the space by making it a smaller bit range too. Maybe even uint8?
```suggestion
        counts[np.newaxis, :, :].astype(np.uint16),
```